std/array.d

```diff
@@ -2517,8 +2517,15 @@ Implements an output range that appends data to an array. This is
 recommended over $(D a ~= data) when appending many elements because it is more
 efficient.
  */
-struct Appender(A)
-if (isDynamicArray!A)
+alias Appender(A) = Buffer!(A, false);
```

Comment:
I don't think adding functionality based on a compile-time boolean is the way to go; why not have a PopFrontAppender with the new functionality and an alias this?